blocks/sharepoint/sharepoint.ts

```diff
@@ -23,6 +23,32 @@ export class Sharepoint {
             label: "Block Description",
           },
         },
+        {
+          ref: "connection_id",
+          component: "SelectInput",
+          componentProps: {
+            label: "Connection",
+            placeholder: "Select a connection",
+            options: async (cbk) => {
+              const response = await cbk.api.get<any[]>(
+                "/public/integrations/sharepoint/overview"
+              );
+              return response
+                ? response
+                    .map(({ connectionId, name, email }) => ({
+                      value: connectionId,
+                      label: `${name} (${email})`,
+                    }))
+                : [];
+            },
+          },
+          validators: [
```

Comment:
You probably need to add 
`whenChanged: (cbk) => {
              cbk.setElementValue("site_id", undefined);
            }`

Reply:
ah... good catch!